system/datatypes/mailinglist.php

```diff
@@ -442,6 +452,77 @@ private function unsubscribeAction(HyphaRequest $request) {
 		return ['redirect', isset($_GET['redirect']) ? $_GET['redirect'] : $this->path()];
 	}
 
+	/**
+	 * @param HyphaRequest $request
+	 * @return null|array
+	 */
+	private function remindAction(HyphaRequest $request) {
+		$code = isset($_GET['code']) ? $_GET['code'] : null;
+		if (null == $code) {
+			notify('error', __('missing-arguments'));
+			return null;
+		}
+
+		$this->xml->lockAndReload();
+
+		$xpath = './/' . self::FIELD_NAME_ADDRESS . '[@' . self::FIELD_NAME_CONFIRM_CODE . '=' . xpath_encode($code) . ' and @' . self::FIELD_NAME_STATUS . '!="' . self::ADDRESS_STATUS_UNSUBSCRIBED . '"]';
+		$address = $this->findAddresses($xpath)->first();
+
+		if (!$address instanceof HyphaDomElement) {
+			$this->xml->unlock();
+			notify('error', __('invalid-code'));
+			return null;
+		}
+
+		if ($address->getAttribute(self::FIELD_NAME_STATUS) !== self::ADDRESS_STATUS_PENDING) {
+			$this->xml->unlock();
+			notify('error', __('ml-no-reminder-address-no-longer-pending'));
+			return null;
+		}
+
+		if ($address->getAttribute(self::FIELD_NAME_REMINDED) != true) {
+			$email = $address->getAttribute(self::FIELD_NAME_EMAIL);
+			$address->setAttribute(self::FIELD_NAME_REMINDED, true);
+			$this->xml->saveAndUnlock();
+
+			$this->sendConfirmationMail($code, $email);
+		}
+
+		notify('success', ucfirst(__('ml-reminder-sent')));
+		return ['redirect', $this->path(self::PATH_ADDRESSES)];
+	}
+
+	/**
+	 * @param HyphaRequest $request
+	 * @return null|array
+	 */
+	private function deleteAddressAction(HyphaRequest $request) {
+		$code = isset($_GET['code']) ? $_GET['code'] : null;
+		if (null == $code) {
+			notify('error', __('missing-arguments'));
+			return null;
+		}
+
+		$this->xml->lockAndReload();
+
+		$xpath = './/' . self::FIELD_NAME_ADDRESS . '[@' . self::FIELD_NAME_CONFIRM_CODE . '=' . xpath_encode($code) . ' and @' . self::FIELD_NAME_STATUS . '!="' . self::ADDRESS_STATUS_UNSUBSCRIBED . '"]';
```

Comment:
This also allows deleting confirmed addresses, which is not what the commit message says. Is that intentional?

Reply:
fixed it